rpm/0091-sailfishos-gecko-Initialise-SVGGeometryProperty-Reso.patch

```diff
@@ -0,0 +1,246 @@
+   float x, y, r;
++  x = y = r = 0.0f;
+   SVGGeometryProperty::ResolveAllAllowFallback<SVGT::Cx, SVGT::Cy, SVGT::R>(
+       this, &x, &y, &r);
+ 
+diff --git a/dom/svg/SVGEllipseElement.cpp b/dom/svg/SVGEllipseElement.cpp
+index f392327fe6c2..b974beef494a 100644
+--- a/dom/svg/SVGEllipseElement.cpp
++++ b/dom/svg/SVGEllipseElement.cpp
+@@ -80,6 +80,7 @@ already_AddRefed<DOMSVGAnimatedLength> SVGEllipseElement::Ry() {
+ /* virtual */
+ bool SVGEllipseElement::HasValidDimensions() const {
+   float rx, ry;
++  rx = ry = 1.0f;
+ 
+   MOZ_ASSERT(GetPrimaryFrame());
+   SVGGeometryProperty::ResolveAll<SVGT::Rx, SVGT::Ry>(this, &rx, &ry);
+@@ -99,6 +100,7 @@ bool SVGEllipseElement::GetGeometryBounds(
+     Rect* aBounds, const StrokeOptions& aStrokeOptions,
+     const Matrix& aToBoundsSpace, const Matrix* aToNonScalingStrokeSpace) {
+   float x, y, rx, ry;
++  x = y = rx = ry = 0.0f;
+ 
+   MOZ_ASSERT(GetPrimaryFrame());
+   SVGGeometryProperty::ResolveAll<SVGT::Cx, SVGT::Cy, SVGT::Rx, SVGT::Ry>(
+@@ -138,6 +140,7 @@ bool SVGEllipseElement::GetGeometryBounds(
+ 
+ already_AddRefed<Path> SVGEllipseElement::BuildPath(PathBuilder* aBuilder) {
+   float x, y, rx, ry;
++  x = y = rx = ry = 0.0f;
+ 
+   SVGGeometryProperty::ResolveAllAllowFallback<SVGT::Cx, SVGT::Cy, SVGT::Rx,
+                                                SVGT::Ry>(this, &x, &y, &rx,
+diff --git a/dom/svg/SVGForeignObjectElement.cpp b/dom/svg/SVGForeignObjectElement.cpp
+index cc9b800fb81c..c7ea5bc649dc 100644
+--- a/dom/svg/SVGForeignObjectElement.cpp
++++ b/dom/svg/SVGForeignObjectElement.cpp
+@@ -80,6 +80,7 @@ gfxMatrix SVGForeignObjectElement::PrependLocalTransformsTo(
+   }
+   // our 'x' and 'y' attributes:
+   float x, y;
++  x = y = 0.0f;
+ 
+   if (GetPrimaryFrame()) {
+     SVGGeometryProperty::ResolveAll<SVGT::X, SVGT::Y>(this, &x, &y);
+@@ -101,6 +102,7 @@ gfxMatrix SVGForeignObjectElement::PrependLocalTransformsTo(
+ /* virtual */
+ bool SVGForeignObjectElement::HasValidDimensions() const {
+   float width, height;
++  width = height = 1.0f;
+ 
+   MOZ_ASSERT(GetPrimaryFrame());
+   SVGGeometryProperty::ResolveAll<SVGT::Width, SVGT::Height>(
+diff --git a/dom/svg/SVGImageElement.cpp b/dom/svg/SVGImageElement.cpp
+index b63a3f1ac1d8..7adb6175a983 100644
+--- a/dom/svg/SVGImageElement.cpp
++++ b/dom/svg/SVGImageElement.cpp
+@@ -249,6 +249,7 @@ bool SVGImageElement::GetGeometryBounds(
+     Rect* aBounds, const StrokeOptions& aStrokeOptions,
+     const Matrix& aToBoundsSpace, const Matrix* aToNonScalingStrokeSpace) {
+   Rect rect;
++  rect.x = rect.y = rect.width = rect.height = 1.0f;
+ 
+   MOZ_ASSERT(GetPrimaryFrame());
+   SVGGeometryProperty::ResolveAll<SVGT::X, SVGT::Y, SVGT::Width, SVGT::Height>(
+@@ -278,6 +279,7 @@ already_AddRefed<Path> SVGImageElement::BuildPath(PathBuilder* aBuilder) {
+ /* virtual */
+ bool SVGImageElement::HasValidDimensions() const {
+   float width, height;
++  width = height = 1.0f;
```

Comment:
Cannot quite get why some floats that are passed to ResolveAll are initialized to 0.0f and some 1.0f. Could @llewelld please elaborate?